src/platform/posix/posix_epdesc.c

```diff
@@ -104,7 +104,7 @@ nni_posix_epdesc_doconnect(nni_posix_epdesc *ed)
 			// Success!
 			nni_posix_pollq_remove(&ed->node);
 			nni_posix_epdesc_finish(aio, 0, ed->node.fd);
-			ed->node.fd = -1;
+			// ed->node.fd = -1;
```

Comment:
Lets just remove it altogether, please.

Reply:
Actually, I believe this code needs to stay the way it was.  Now that we are *removing* the pollq, the value of the fd is -1, and this ensures that we do not attempt to reuse the same descriptor for connect, but instead allocate a new one.   I'm wondering if this is the source of your grief about remove followed by arm...